flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -93,6 +137,23 @@ public JobExecutionResult execute(StreamGraph streamGraph) throws Exception {
         }
     }
 
+    private void checkNotAllowedConfigurations(StreamGraph streamGraph)
+            throws FatalProgramInvocationException {
+        errorMessages.addAll(collectNotAllowedConfigurations());
+        if (!errorMessages.isEmpty()) {
+            // HACK: We shortcut the StreamGraph to jobgraph translation because we already
+            // know that the job needs to fail and can derive the jobId.
```

Comment:
The job fails before being submitted, there is no need to generate jobId here. Also wouldn't this break with multiple job submission (we support that in non-ha setups)?

Reply:
Yes, you are right in the case of multiple executes we will only capture the first one https://github.com/apache/flink/pull/17995#discussion_r762959216 .

I would see exception handling for multiple jobs out of scope for this PR because therefore we probably have to rethink how the Application mode interacts with the cluster components.